libxrdp/xrdp_caps.c

```diff
@@ -521,6 +522,7 @@ xrdp_caps_process_codecs(struct xrdp_rdp *self, struct stream *s, int len)
     int i1;
     char *codec_guid;
```

Comment:
May need `const struct guid *codec_guid` instead if we use the function in guid.c. Also the cast will need to be changed.

Reply:
I will do this later in this PR. Can you have a look again at other points?

Reply:
The code looks good to me now.

I've read the link you've provided, and I don't think there's any difference between Microsoft's interpretation of a GUID and RFC4122. In other words, I think you can just call get rid of the existing `guid_to_str()` and replace it with your new function completely. The existing uses of the function should be fine.

Reply:
Is it fine to add hyphens for other use cases of  `guid_to_str()`?

Reply:
>  The existing uses of the function should be fine.

Ah, you've already mentioned that.

Reply:
@matt335672 I've replaced your `guid_to_str()` with my `ms_guid_to_str()`.  Can you have a look again?